trio/_core/_multierror.py

```diff
@@ -174,21 +180,21 @@ class MultiError(BaseException):
 
     """
 
-    def __init__(self, exceptions):
-        # Avoid recursion when exceptions[0] returned by __new__() happens
-        # to be a MultiError and subsequently __init__() is called.
-        if hasattr(self, "exceptions"):
-            # __init__ was already called on this object
-            assert len(exceptions) == 1 and exceptions[0] is self
+    def __init__(self, exceptions, *, _collapse=True):
+        # Avoid double initialization when _collapse is True and exceptions[0] returned
+        # by __new__() happens to be a MultiError and subsequently __init__() is called.
+        if _collapse and getattr(self, "exceptions", None) is not None:
+            # This exception was already initialized.
             return
-        self.exceptions = exceptions
 
-    def __new__(cls, exceptions):
+        super().__init__("multiple tasks failed", exceptions)
```

Comment:
`"multiple tasks failed"` might not be true, if `strict_exception_groups=True` - do we want to change the message?

Reply:
Then we'd have to check the entire exception hierarchy to see if they collapse to a single exception, wouldn't we?

Reply:
Hmm, yeah, that's a good argument against.